### ISSUE TITLE
Document outline and block navigation in text mode

### ISSUE THREAD

**Issue by gziolo** [WordPress/gutenberg#12157]
Clicking on links in the document outline and block navigation can't really work in text mode. Should we disable those features or convert links to plain text?

![screen shot 2018-11-21 at 09 28 20](https://user-images.githubusercontent.com/699132/48828493-23f24480-ed70-11e8-8435-748df1c04383.png)
![screen shot 2018-11-21 at 09 28 29](https://user-images.githubusercontent.com/699132/48828494-23f24480-ed70-11e8-9f70-59dbf958e87d.png)

/cc @jasmussen

**Comment by jasmussen** [comment]
Can we just disable those button in the text editor?

**Comment by gziolo** [comment]
Sure, @tofumatt or @afercia  should they be marked as disabled or can be removed altogether when Code editor is active?

**Comment by afercia** [comment]
I'd vote for disabling them. Please consider there's some inconsistency in the way these controls get disabled, I've tried to improve consistency in https://github.com/WordPress/gutenberg/pull/12073

**Comment by nicolad** [comment]
Had disabled button, please confirm if that it should not be deleted.
https://github.com/WordPress/gutenberg/pull/12185

**Comment by jasmussen** [comment]
Let's go with disabling them for now, then we can revisit if it becomes pressing. 👍 👍

**Comment by afercia** [comment]
I'd suggest to address this together with #12073, which takes a different approach to "disable" those buttons.

**Comment by afercia** [comment]
Discussed during today's accessibility bug scrub, and agreed to recommend the approach taken in #12073 to "disable" those buttons. Removing the Needs Accessibility Feedback label, as feedback has been given.

**Comment by gziolo** [comment]
Navigation was fixed in #12185.

**Comment by afercia** [comment]
@gziolo what's left here? Can this be closed?

**Comment by gziolo** [comment]
Document navigation still needs to be adjusted.

There is also one thing that needs to be standardized:

> in my testing (for block navigation) there is a subtle difference between `disabled` and `aria-disabled` - in case of using `disabled` you never have to tab through the disabled element which seems to be reasonable but differs from other buttons like `redo` or `undo`.

**Comment by afercia** [comment]
Hm will have a look.

`aria-disabled` doesn't do anything in terms of interaction and focus. It just communicates to assistive technologies a certain element is disabled. To avoid focus losses, some buttons just `noop` and use `aria-disabled` so they're still focusable but they don't do anything.

**Focusability of disabled controls**
https://www.w3.org/TR/wai-aria-practices-1.1/#kbd_disabled_controls

**Comment by afercia** [comment]
Yep, the change introduced in #12185 is inconsistent with the behavior of the other buttons. It should use `aria-disabled` and `noop`, instead of the HTML attribute `disabled`. Will try a PR, looking also at Document navigation.

**Comment by afercia** [comment]
I see there's also a warning:
```
Warning: React does not recognize the `isSelected` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `isselected` instead. If you accidentally passed it from a parent component, remove it from the DOM element.
    in button (created by ForwardRef(Button))
    in div (created by BlockNavigationList)
    in li (created by BlockNavigationList)
    ...
```

**Comment by afercia** [comment]
Turns out the `isSelected` prop is there because these where previously `MenuItem` components. Now, they're buttons and the prop should be removed.
